docs/contributing/code-review.md

```diff
@@ -0,0 +1,121 @@
+  Because a PR is a reflection of a _deploy_, the author of the PR should also be the one that merges it, since only they know what infrastructure has actually been deployed.
+- **Explicitly list a back-up if you must step away**. (REQUIRED)
+  PR authors are responsible for the infrastructure changes that they make.
+  You should generally only make a change to live infrastructure if you'll have the bandwidth to ensure it is fixed if something goes wrong.
+  If for some reason you **must** step away, make it clear who else is responsible for shepherding the PR.
+- **Be careful when self-merging without review**.
+  Because changing active infrastructure is potentially confusing or disruptive to users, be extra careful if you self-merge without a review approval.
+  Consider whether your commit will cause a change that might be destructive and ask if you _really_ need to merge now or can wait for review.
+  That said, sometimes the only way to understand the impact of a change is to merge and see how things go, so use your best judgment!
+
+  :::{admonition} Common things that often _don't_ need an approval
+  - Updating admin users for a hub
+  - Changing basic hub configuration such as the URL of its landing page image
+  - Updating the user image of a hub.
+  :::
+
+(infrastructure:review:terraform)=
+## Terraform changes
+
+[Terraform](tools:terraform) is used for directly provisioning and modifying cloud infrastructure (as opposed to just deploying applications on pre-existing cloud infrastructure).
+This lets us treat our infrastructure as code, so we can use regular code quality practices (code review, linters, extensive documentation, etc) to keep the quality of our infrastructure high.
+However, it comes with a few caveats:
+
+1. Locally and iteratively testing terraform is impossible - as we
+   develop it, the infrastructure is deployed / modified / destroyed
+   as we go.
+2. Behavior of terraform code differs based on the current
+   state of the infrastructure (as represented in the
+   [state file](https://www.terraform.io/docs/language/state/index.html)).
+3. While `terraform plan` can tell us what terraform *thinks* it is
+   going to do, it might not always succeed. Permission errors or
+   organizational restrictions can cause it to fail. Timeouts can happen.
+   Runtime errors pop up.
+
+Thanks to all these, it is impossible to either:
+
+- Accurately gauge the impact of a terraform code change just from looking at the text
+- Automatically deploy it via continuous deployment
+
+As a result, we have some workflows that are specific to using Terraform, described below.
+
+### Changes to running infrastructure
+
+In this scenario, there is already-running infrastructure and you are deploying changes to it.
+
+For modifications to *existing infrastructure*, the suggested workflow is:
+
+
+3. 
+4. After the change has been successfully performed with `terraform apply`,
+   self-merge your PR.
+
+
```

Comment:
Are these lines to be deleted in favor of the `To deploy changes, follow these steps:` section just below?

```suggestion
```